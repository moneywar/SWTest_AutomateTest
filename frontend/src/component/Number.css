.warpper {
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  left: 0;
  width: 100vw;
}

.box {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 250px;  
  height: 100px; 
  margin: 20px;
  background-color: aquamarine;
  border-radius: 5px;
  box-shadow: 0 0 10px rgba(0,0,0,0.1);
} 

