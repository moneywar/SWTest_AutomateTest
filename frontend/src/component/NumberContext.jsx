import React, { createContext, useState } from 'react';

export const NumberContext = createContext();

export const NumberProvider = ({ children }) => {
    const [number, setNumber] = useState(0); // Shared state for the number

    return (
        <NumberContext.Provider value={{ number, setNumber }}>
            {children}
        </NumberContext.Provider>
    );
};