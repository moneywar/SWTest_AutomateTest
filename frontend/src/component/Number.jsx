import React, { useEffect, useContext } from 'react';
import { NumberContext } from './NumberContext';
import './Number.css';

function App() {
  const { number, setNumber } = useContext(NumberContext); // Access the shared state and updater function

  useEffect(() => {
    fetch('http://localhost:3001/get-number')
      .then((response) => response.json())
      .then((data) => {
        setNumber(data.number); // Use the correct property name
        console.log(data.number);
      })
      .catch((err) => {
        console.error('Error fetching the number:', err);
        alert('Failed to fetch the number from the server.');
      });
  }, []);

  return (
    <div className='warpper'>
      <div className='box'> 
        <p>{number}</p>
      </div>
    </div>
  );
}

export default App;