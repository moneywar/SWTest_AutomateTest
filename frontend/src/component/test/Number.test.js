import { render } from '@testing-library/react';
import { NumberContext } from '../NumberContext';
import Number from '../Number';

const mockContextValue = {
  number: 5,
  setNumber: jest.fn(),
};

describe('Number component snapshot testing', () => {
  it('matches snapshot with mock context', () => {
    const { asFragment } = render(
      <NumberContext.Provider value={mockContextValue}>
        <Number />
      </NumberContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
