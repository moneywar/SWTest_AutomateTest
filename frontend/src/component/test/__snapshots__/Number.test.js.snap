// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component snapshot testing matches snapshot with mock context 1`] = `
<DocumentFragment>
  <div
    class="warpper"
  >
    <div
      class="box"
    >
      <p>
        5
      </p>
    </div>
  </div>
</DocumentFragment>
`;
